34.md

```diff
@@ -0,0 +1,79 @@
+NIP-34
+======
+
+`git` stuff
+-----------
+
+`draft` `optional`
+
+This NIP defines all the ways code collaboration using and adjacent to [`git`](https://git-scm.com/) can be done using Nostr.
+
+## Repository announcements
+
+Git repositories are hosted in Git-enabled servers, but their existence can be announced using Nostr events, as well as their willingness to receive patches, bug reports and comments in general.
+
+```jsonc
+{
+  "kind": 30617,
+  "content": "",
+  "tags": [
+    ["d", "<repo-id>"],
```

Comment:
A word on this, please. I can't find a repo-id definition

Reply:
It's just an arbitrary string, as in all other `d` tags.

Reply:
Wouldn't it makes sense to prescribe this? If we used the initial commit hash for example, finding/detecting forks would be easier. `fiatjaf:30617:helloWorld` might be a very different repo than `giszmo:30617:helloWorld` but `giszmo:30617:<sha256-of-bitcoin-root-commit>` ... would hopefully match between all forks.

Reply:
I like this idea a lot - something github doesn't do and we could - that allows people to find all the forks.  But this particular event contains data that would differ between forks.

Reply:
Some OSS products which fork has become the official developer because the original author retired the development.

https://github.com/ctrlpvim/ctrlp.vim

Reply:
Pull request events will need to be very specific about which repo they are targeting. Using the initial commit hash as the `d` tag makes it impossible to use this as the identifier for pull requests.

Reply:
Why not make it the root commit id? For forks it could be the id of the earliest unique commit. The repository COULD contain a config file which specifies this id so that clients MAY access the config to automatically identify the correct id.